SU2_CFD/src/solvers/CTurbSASolver.cpp

```diff
@@ -509,18 +584,54 @@ void CTurbSASolver::BC_Isothermal_Wall(CGeometry *geometry, CSolver **solver_con
 
     if (geometry->node[iPoint]->GetDomain()) {
 
-      /*--- Get the velocity vector ---*/
-      for (iVar = 0; iVar < nVar; iVar++)
-        Solution[iVar] = 0.0;
+      if (!rough_wall) {
 
-      nodes->SetSolution_Old(iPoint,Solution);
-      LinSysRes.SetBlock_Zero(iPoint);
+        /*--- Get the solution vector ---*/
+        for (iVar = 0; iVar < nVar; iVar++)
+          Solution[iVar] = 0.0;
```

Comment:
`Solution[0] = 0.0` instead, the same assumption (nVar==1) is made when setting the diagonal of the matrix to 1.